perfkitbenchmarker/providers/openstack/os_disk.py

```diff
@@ -12,107 +12,154 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import json
 import logging
-import time
-
 
+from perfkitbenchmarker import disk
 from perfkitbenchmarker import errors
 from perfkitbenchmarker import flags
-from perfkitbenchmarker import disk
+from perfkitbenchmarker import vm_util
 from perfkitbenchmarker.providers.openstack import utils as os_utils
 
 FLAGS = flags.FLAGS
 
+REMOTE_BOOT_DISK_SIZE_GB = 50
+
+
+def CreateVolume(resource, name, image=None):
+  """Creates a remote (Cinder) block volume."""
+  vol_cmd = os_utils.OpenStackCLICommand(resource, 'volume', 'create', name)
+  vol_cmd.flags['availability-zone'] = resource.zone
+  if image:
+    vol_cmd.flags['image'] = image
+    vol_cmd.flags['size'] = GetImageMinDiskSize(resource, image)
+  if FLAGS.openstack_volume_size:
+    vol_cmd.flags['size'] = FLAGS.openstack_volume_size
+  else:
+    vol_cmd.flags['size'] = REMOTE_BOOT_DISK_SIZE_GB
+  stdout, _, _ = vol_cmd.Issue()
+  vol_resp = json.loads(stdout)
+  return vol_resp
+
+
+def GetImageMinDiskSize(resource, image):
+  """Returns minimum disk size required by the image."""
+  image_cmd = os_utils.OpenStackCLICommand(resource, 'image', 'show', image)
+  stdout, _, _ = image_cmd.Issue()
+  image_resp = json.loads(stdout)
+  volume_size = max((int(image_resp['min_disk']), REMOTE_BOOT_DISK_SIZE_GB,))
+  return volume_size
+
+
+def DeleteVolume(resource, volume_id):
+  """Deletes a remote (Cinder) block volume."""
+  vol_cmd = os_utils.OpenStackCLICommand(resource, 'volume', 'delete',
+                                         volume_id)
+  del vol_cmd.flags['format']  # volume delete does not support json output
+  vol_cmd.Issue()
+
+
+@vm_util.Retry(poll_interval=1, max_retries=-1, timeout=300, log_errors=False,
```

Comment:
For the most part, this operation is really quick (< 5 seconds for empty volumes) which is why I have the small `poll_interval`, but thinking about it, when we create a boot volume with an image it might take longer, but it will depend on the backend implementation they configured to use (LVM, Ceph, etc). I can bump it up to 5 seconds, that should be good enough for most cases.